styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java

```diff
@@ -44,6 +50,35 @@ private KubernetesPodEventTranslator() {
   private static final Predicate<ContainerStatus> IS_STYX_CONTAINER =
       (cs) -> KubernetesDockerRunner.STYX_RUN.equals(cs.getName());
 
+  @AutoMatter
+  private interface TerminationLogMessage {
+    String componentId();
+    String workflowId();
+    String parameter();
+    String executionId();
+    String event();
+    int exitCode();
+  }
+
+  private static int getExitCode(Pod pod, ContainerStatus status) {
+    if ("true".equals(pod.getMetadata().getAnnotations().get(DOCKER_TERMINATION_LOGGING_ANNOTATION))) {
+      try {
+        final TerminationLogMessage message = Json.OBJECT_MAPPER.readValue(
+            status.getState().getTerminated().getMessage(),
```

Comment:
If user specified the flag, but forgot (somehow) to write termination message, will we get NPE here?